src/hooks/useMedia.tsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react"
+
+export function useMedia<Value extends unknown>(
+  queries: string[],
+  values: Value[],
+  defaultValue: Value
+) {
+  const [currentValue, setCurrentValue] = useState(defaultValue)
+
+  useEffect(() => {
+    if (typeof window === "undefined") return
+
+    const mediaQueries = queries.map((query) => window.matchMedia(query))

```

Comment:
I think it can be fixed the same way other hook does it:

`const windowGlobal = typeof window !== "undefined" && window`

Reply:
Aaaah. Good Idea.